### PR TITLE
Add hook to setup clang isystem path for some cases

### PR DESCRIPTION
**Pull request by IuriiKobein**

### CODE REVIEW

File: lib/core/sys/setup.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#include <cstdlib>
+
+namespace {
+std::unique_ptr<std::string> CLANG_SYSROOT_DIR;
+std::unique_ptr<std::string> CLANG_ISYSTEM_OPT;
+
+constexpr char EB_ROOT_CLANG[] = "EBROOTCLANG";
+
+__attribute__((constructor)) void initClangEnvVars() {
+    CLANG_SYSROOT_DIR = std::make_unique<std::string>();
```
Comment (kchabSS) [inline lib/core/sys/setup.cpp:16]: Strings are already pointers to data (internally).
If you wish to make it properly, you may change `getSysRoot` and `getISystemOpt` return references to a string, that is statically allocated inside.

File: lib/attributes/backend/cuda/translation_unit.cpp [hunk 2]
```diff
@@ -15,7 +15,9 @@ const std::string CUDA_PL_PRIM_INC = "<cuda_pipeline_primitives.h>";
 
 std::vector<std::string_view> getBackendHeader(SessionStage& s) {
     auto defines = s.getSession().getInput().defines;
-    auto hasAsyncMode = std::find(defines.begin(), defines.end(), "USE_ASYNC_READ");
+    auto hasAsyncMode = std::find_if(defines.begin(), defines.end(), [](const std::string& d) {
+        return d.find("USE_ASYNC_READ") != std::string::npos;
```
Comment (kchabSS) [inline lib/attributes/backend/cuda/translation_unit.cpp:19]: This code may react to `NOTUSE_ASYNC_READ` which is not an intended behavior.

**Review by kchabSS — CHANGES_REQUESTED** [review]
Please static/unique_ptr part of code.

**Review by kchabSS — APPROVED** [review]

File: lib/attributes/backend/cuda/translation_unit.cpp (hunk 2 above)
Comment (IuriiKobein) [reply lib/attributes/backend/cuda/translation_unit.cpp:19]: It will be removed after Victor's PR.

File: lib/core/sys/setup.cpp (hunk 1 above)
Comment (IuriiKobein) [reply lib/core/sys/setup.cpp:16]: Initially it was done in this manner, however it does not work.
Do not have time to dig into so used ptr instead.

**Pull request merged**